System.IO.FileSystem/File.cs

```diff
@@ -166,39 +142,89 @@ public static void Delete(string path)
         /// Determines whether the specified file exists.
         /// </summary>
         /// <param name="path">The file to check.</param>
-        /// <returns></returns>
+        /// <returns><c>true</c> if the file exists; otherwise <c>false</c>.</returns>
         public static bool Exists(string path)
         {
             return ExistsNative(Path.GetDirectoryName(path), Path.GetFileName(path));
         }
 
+        /// <summary>
+        /// Gets the <see cref="FileAttributes"/> of the file on the path.
+        /// </summary>
+        /// <param name="path">The path to the file.</param>
+        /// <exception cref="IOException"><paramref name="path"/> cannot be not found.</exception>
+        public static FileAttributes GetAttributes(string path)
+        {
+            if (!Exists(path))
+            {
+                throw new IOException(string.Empty, (int)IOException.IOExceptionErrorCode.FileNotFound);
+            }
+
+            var attributes = GetAttributesNative(path);
+
+            if (attributes == 0xFF)
+            {
+                throw new IOException(string.Empty, (int)IOException.IOExceptionErrorCode.FileNotFound);
+            }
+
+            return (FileAttributes)attributes;
+        }
+
+        /// <summary>
+        /// Returns the date and time the specified file or directory was last written to.
+        /// </summary>
+        /// <param name="path">
+        /// The file or directory for which to obtain write date and time information.
+        /// </param>
+        /// <returns>
+        /// A <see cref="DateTime" /> structure set to the last write date and time for the specified file or directory.
+        /// </returns>
+        /// <exception cref="IOException"><paramref name="path"/> cannot be not found.</exception>
+        public static DateTime GetLastWriteTime(string path)
+        {
+            if (!Exists(path))
+            {
+                throw new IOException(string.Empty, (int)IOException.IOExceptionErrorCode.FileNotFound);
+            }
+
+            return GetLastWriteTimeNative(path);
+        }
+
         /// <summary>
         /// Moves a specified file to a new location, providing the option to specify a new file name.
         /// </summary>
-        /// <param name="sourceFileName">The name of the file to move. Absolute path.</param>
+        /// <param name="sourceFileName">The name of the file to move. Must be an absolute path.</param>

```

Comment:
Being an absolute path it's a requirement of our API? 
Asking because this is different from the full .NET.

Reply:
This is different from the full API and I was just following the existing documentation. Let me run a quick test.

Reply:
Confirmed that relative path will not work for either parameter at this point. 

For a relative source parameter we immediately fail the `Exists` check and would encounter other errors downstream. I think the root issue here is, unless I'm missing something, nanoFramework does not have a concept of the current directory for the running application.

For a relative destination parameter we fail the `Path.GetPathRoot` check to determine if the source and destination are on the same volume. This could be resolved by checking if the destination is a relative path and combining the root / directory from the source to it. That being said if we did want to allow a relative destination then I would prefer to do it with or after my refactoring of `Path` (next on my plate :P)

Reply:
Got you! Thanks for checking. Let's make this clear in the comments.